rdr_service/api/nph_participant_api_schemas/schema.py

```diff
@@ -252,10 +264,22 @@ class Meta:
     def resolve_participant(root, info, nph_id=None, sort_by=None, limit=None, off_set=None, **kwargs):
         with database_factory.get_database().session() as sessions:
             logging.info('root: %s, info: %s, kwargs: %s', root, info, kwargs)
-            query = sessions.query(ParticipantSummaryModel, Site, ParticipantMapping) \
-                .join(Site, ParticipantSummaryModel.siteId == Site.siteId) \
-                .join(ParticipantMapping,
-                      ParticipantSummaryModel.participantId == ParticipantMapping.primary_participant_id)
+            pm2 = aliased(PairingEvent)
+            query = sessions.query(ParticipantSummaryModel, Site, nphSite, ParticipantMapping
+                                   ).join(Site, ParticipantSummaryModel.siteId == Site.siteId
+                                          ).join(ParticipantMapping,
+                                                 ParticipantSummaryModel.participantId
+                                                 == ParticipantMapping.primary_participant_id
+                                                 ).join(PairingEvent, PairingEvent.participant_id
+                                                        == ParticipantMapping.ancillary_participant_id
+                                                        ).outerjoin(pm2, and_(PairingEvent.participant_id
+                                                                              == pm2.participant_id,
+                                                                              PairingEvent.event_type_id
+                                                                              == pm2.event_type_id,
+                                                                              PairingEvent.event_authored_time
+                                                                              < pm2.event_authored_time)
+                                                                    ).join(nphSite, nphSite.id == PairingEvent.site_id
+                                                                           ).filter(pm2.id.is_(None))
```

Comment:
I hadn't thought about this at the time we talked about the query, but since this query is only to be ran on NPH, we probably want to add another filter condition here so that we don't need to update the query later when we add another ancillary study. It would look like:
.filter(ParticipantMapping.ancillary_study_id == 2)

Reply:
included another filter as Josh suggested.